specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,209 @@
+Hit Test Kind
+===
+
+# Background
+The ability to ask WebView2 for information on a hit test is an important feature for 
+visually hosted apps. It can be used to support features related to mouse activity in 
+regions that are traditionally Non-Client areas. Features like Draggable Regions, 
+Resize, Minimize/Maximize functionality, and more.
+
+
+# Examples
+## Draggable regions (HWND)
+Draggable regions are marked regions on a webpage that will move the app window when the 
+user clicks and drags on them. It also opens up the system menu if right-clicked. This 
+API is designed to provide draggable regions support for apps that are directly hosting 
+the CoreWebView2 via visual hosting and for UI framework WebView2 controls that use visual 
+hosting, like the WinUI2 and WinUI3 WebView2 control.If your app uses windowed hosting, 
+draggable regions are supported by default, no need to follow this guide.
+
+NOTE: Your app may already be forwarding mouse messages to WebView2, if so the bulk 
+of the code below is likely already in your app. One thing you need to do specifically 
+for this feature is to add the check for w_param == HTCAPTION as a clause for capturing 
+the mouse and forwarding messages. It is necessary for the system menu to show. If you 
+choose not to include this you would have to handle the logic for showing the system menu 
+yourself.
+
+NOTE: This API is not currently supported for environments that use a CoreWindow like UWP
+
+#### Win32 C++
+```cpp
+LRESULT CALLBACK WndProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+....
+// The following code forwards mouse messages to the WebView2 - Start
+
+// handle the range of mouse messages & WM_NCRBUTTONDOWN
+if (message >= WM_MOUSEFIRST && message <= WM_MOUSELAST 
+    || message == WM_NCRBUTTONDOWN || message == WM_NCRBUTTONUP) 
+{
+    POINT point {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+    // IsPointInWebview is an example of a function which your app can implement that checks if 
+    // the mouse message point lies inside the rect of the visual that hosts the WebView2
+    if (IsPointInWebView(point) || message == WM_MOUSELEAVE) 
```

Comment:
This will send every `WM_MOUSELEAVE` to the WebView2, even if it was not previously within its bounds. Intended?

Reply:
Also note that WM_MOUSELEAVE is only generated if TrackMouseEvent has been called.

Reply:
Add a comment explaining why mouse leave doesn't have to do bounds check - and what code is sort of missing that is in the WebView2 API sample. And add a link to the full sample app.

Reply:
Removed the WM_MOUSELEAVE condition. It is not withing the scope of this new API and adding documentation for it will involve bloating the example code.